### PR TITLE
Exclude test config

### PR DESCRIPTION
**Pull request by jonfox**
Currently when project dependencies are worked out in a multi-project build, there is no check to ensure that we are only including projects with a compile dependency, i.e. those with `.dependsOn(module1)` or `.dependsOn(module1, "compile->...")`

This fixes that issue and filters out projects that don't have compile dependencies. I think this is what is being referred to in issue #142 

There was also a problem with the build, sbt would throw on startup. This is fixed in the first commit.

### CODE REVIEW

**Comment by jonfox** [comment]
Also it looks like Travis has not been building for a while: https://travis-ci.org/xerial/sbt-pack/builds

File: src/sbt-test/sbt-pack/exclude-test-config/build.sbt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  scalaVersion := "2.12.6",
+  version := "0.1",
+  crossPaths := false
+)
+
+lazy val module1 =
+  project
+    .settings(commonSettings)
+    .settings(
+      libraryDependencies ++= Seq(
```
Comment (xerial) [inline src/sbt-test/sbt-pack/exclude-test-config/build.sbt]: Can we reduce the number of dependencies to just one to simplify the test case?

**Comment by xerial** [comment]
Thanks for the fix. I guess you copied a test case from the other test. To make it easy to understand, I'd like to simplify it (by using a single dependency). 

If this is fixed, it's ok to merge. I'll check travis ci configuration later.

File: src/sbt-test/sbt-pack/exclude-test-config/build.sbt (hunk 1 above)
Comment (jonfox) [reply src/sbt-test/sbt-pack/exclude-test-config/build.sbt]: Yes I'll do that. As you note, I started with a copy of another test

**Comment by jonfox** [comment]
I have also updated the fix build commit as I had problems when I hooked my fork up to Travis:
- it looks like the `.travis.yml` schema has changed and the existing file was invalid
- the Oracle JDK 8 is no longer available on Travis so I switched to OpenJDK
- I updated paulp's sbt script
- the previous Travis config was not in fact building against sbt 0.13, rather it was just using the `build.properties` version. When I tried to use `-sbt-version`  with 0.13 I got errors due to current versions of some sbt plugins not being published for 0.13, so am just using the `build.properties` version now

This seems to have kicked off a Travis build on this repo which looks like it's passed now.

**Comment by xerial** [comment]
LGTM. Thanks for the contribution!

**Pull request merged**